.github/workflows/test.yml

```diff
@@ -128,44 +124,56 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        dbimage:
+        postgres-formula:
           # Only test one version on macOS, let Linux do the rest
           - postgresql@14
-        dbauth:
+        postgres-auth:
           # Only test one auth method on macOS, Linux tests will cover the others
           - scram-sha-256
-        xcode:
-          - latest-stable
+        xcode-version:
+          - '~14.3'
+          - '15.0-beta'
     runs-on: macos-13
     env:
       POSTGRES_HOSTNAME: 127.0.0.1
       POSTGRES_USER: 'test_username'
       POSTGRES_PASSWORD: 'test_password'
       POSTGRES_DB: 'postgres'
-      POSTGRES_HOST_AUTH_METHOD: ${{ matrix.dbauth }}
+      POSTGRES_AUTH_METHOD: ${{ matrix.postgres-auth }}
       POSTGRES_SOCKET: '/tmp/.s.PGSQL.5432'
-      POSTGRES_VERSION: ${{ matrix.dbimage }}
+      POSTGRES_FORMULA: ${{ matrix.postgres-formula }}
     steps:
       - name: Select latest available Xcode
         uses: maxim-lobanov/setup-xcode@v1
         with:
-          xcode-version: ${{ matrix.xcode }}
+          xcode-version: ${{ matrix.xcode-version }}
       - name: Install Postgres, setup DB and auth, and wait for server start
         run: |
-          export PATH="$(brew --prefix)/opt/${POSTGRES_VERSION}/bin:$PATH" PGDATA=/tmp/vapor-postgres-test
-          (brew unlink postgresql || true) && brew install "${POSTGRES_VERSION}" && brew link --force "${POSTGRES_VERSION}"
-          initdb --locale=C --auth-host "${POSTGRES_HOST_AUTH_METHOD}" -U "${POSTGRES_USER}" --pwfile=<(echo "${POSTGRES_PASSWORD}")
+          export PATH="$(brew --prefix)/opt/${POSTGRES_FORMULA}/bin:$PATH" PGDATA=/tmp/vapor-postgres-test
+          (brew unlink postgresql || true) && brew install "${POSTGRES_FORMULA}" && brew link --force "${POSTGRES_FORMULA}"
+          initdb --locale=C --auth-host "${POSTGRES_AUTH_METHOD}" -U "${POSTGRES_USER}" --pwfile=<(echo "${POSTGRES_PASSWORD}")
           pg_ctl start --wait
         timeout-minutes: 2
       - name: Checkout code
         uses: actions/checkout@v3
       - name: Run all tests
         run: swift test
 
+  code-coverage:
+    runs-on: ubuntu-latest
+    container: swift:jammy
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Run unit tests for coverage data
+        run: swift test --filter=^PostgresNIOTests --enable-code-coverage
+      - name: Upload coverage data
+        uses: vapor/swift-codecov-action@v0.2
+    
```

Comment:
why did you move code cov into its own job? why isn't this part of unit testing anymore?

Reply:
Because I did some digging into what actually results from providing the code coverage from every unit test variant. The results are always identical (except where there are Swift- or DB-version dependent semantics that are touched by unit tests, which is extremely rare right across all of Vapor). In short, it slows down each individual variant (both the build _and_ the execution, enough to be easily measurable in some cases) and provides nowhere near enough valuable data to be worth the trouble even in CI. Splitting it out to its own run adds another job, yes, but that job doesn't need to have TSan enabled; in most repos the end result is the overall time taken to run CI goes _down_, despite using the additional queue slot.

And on a minor note, Codecov's backend seems to appreciate not having to process 20 identical uploads for every single push to a PR; the upload failure rate drops from ~4% to literally zero (at least as observed thus far). On, and for good measure, the additional env vars that were being included with the uploads don't seem to be visible anywhere in Codecov's interface, so even where there _were_ differences between coverage on a per-variant basis, you couldn't tell which was which anyway.